slemicro/get_kubeconfig.sh

```diff
@@ -126,8 +126,13 @@ else
 		while ! [ "$(curl -s -w '%{http_code}' -k -o /dev/null https://${RANCHERHOSTNAME}/healthz)" -eq 200 ]; do sleep 10; done
 	fi
 	
-	# Login
-	TOKEN=$(curl -sk -X POST https://${RANCHERHOSTNAME}/v3-public/localProviders/local?action=login -H 'content-type: application/json' -d "{\"username\":\"admin\",\"responseType\": \"token\", \"password\": \"${RANCHERFINALPASSWORD}\"}" | jq -r .token)
+	TOKEN="null"
+	# Prevent a race condition while the rancher bootstrap is being applied
+	while [ ${TOKEN} == "null" ]; do
```

Comment:
Good call. We need to add it to every curl-loop :D any suggestions?

Reply:
One option is by setting
```
   --connection-timeout 
   --max-time 
```
on the curl and then you do not need the while loop, but i'm not sure of the call can return a response without a TOKEN, needs to be tested

Another option is to implement a timeout on the loop. Something from here can be implemented:
https://kb.novaordis.com/index.php/Bash_Timeout_Loops

Reply:
I've been thinking about this and I think it is better to keep it infinite... I mean, if we put a timeout and it is reached it is because it is still working on it or something happened... and for those cases there is no difference if it fails or if it continues to try as this is a independent process.
Let me know what you think.

Reply:
My opinion is that the timeouts are a good practice because they are a way for catching issues with the setup.
For example, if we use here some reasonable timeout (5 or 10min) for example it will be a nice notification for the user that if it fails, something is not fine with the setup and he has to check that. Otherwise, he will not know if 5 min for example is too much for the setup or if something is broken.

But if you think that in this case, this is not relevant, I will comply with that